core/trino-main/src/main/java/io/trino/execution/StageStateMachine.java

```diff
@@ -338,7 +337,7 @@ public BasicStageStats getBasicStageStats(Supplier<Iterable<TaskInfo>> taskInfos
             internalNetworkInputDataSize += taskStats.getInternalNetworkInputDataSize().toBytes();
             internalNetworkInputPositions += taskStats.getInternalNetworkInputPositions();
 
-            if (fragment.getPartitionedSourceNodes().stream().anyMatch(TableScanNode.class::isInstance)) {
+            if (fragment.containsTableScanNode()) {
```

Comment:
Since fragment is same, you could compute this once before the for loop instead of adding a field to PlanFragment

Reply:
The code `fragment.getPartitionedSourceNodes().stream().anyMatch(TableScanNode.class::isInstance)`
will be called many times (depends on the number of the concurrent stage) in `QueryStateMachine` and `StageStateMachine` if someone open and stay at the cluster webpage.

So I think is better that make 'containsTableScanNode' as a property of 'Fragment' to prevent repeated calculation, and better for codestyle also.